mllib/src/test/scala/org/apache/spark/ml/clustering/BisectingKMeansSuite.scala

```diff
@@ -29,9 +29,12 @@ class BisectingKMeansSuite
   final val k = 5
   @transient var dataset: Dataset[_] = _
 
+  @transient var sparseDataset: Dataset[_] = _
+
   override def beforeAll(): Unit = {
     super.beforeAll()
     dataset = KMeansSuite.generateKMeansData(spark, 50, 3, k)
+    sparseDataset = KMeansSuite.generateSparseData(spark, 100, 1000, k, 42)
```

Comment:
I can only repro the issue with very sparse data where number of columns is around 1k. I was able to reduce the number of rows to be only 10 however.  I hope that is a small enough dataset.